src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -234,42 +319,32 @@ void mf::XWaylandSurface::dirty_properties()
     props_dirty = true;
 }
 
-void mf::XWaylandSurface::set_surface(WlSurface* wl_surface)
+void mf::XWaylandSurface::set_wl_surface(WlSurface* wl_surface)
```

Comment:
It seems rather strange to have a `set_wl_surface()` method that doesn't set a `wl_surface` property.

I suspect we need to cache the surface for use in `create_scene_surface_if_needed()`, but the naming seems strange anyway as "setting" isn't the primary functionality here.

Reply:
Conceptually, this function does set the `WlSurface` that this `XWaylandSurface` will use. The oddity is that the `wl_surface` is not needed directly outside this function, so we use it to initialize some objects and then forget it.

The motivation for doing it this way is that a `WlSurface` can only be safely used on the Wayland thread, so we do everything we need with it in one place. `create_scene_surface_if_needed()` can later be run on any thread.

Open to changes in the name, but any potential name should indicate that after the function completes the `XWaylandSurface` is in a state such that it will use the given `WlSurface` when required, and to me "setting" is the obvious terminology for that.

Reply:
Update: we settled on `attach_wl_surface()`